src/app/static/src/app/store/surveyAndProgram/actions.ts

```diff
@@ -332,7 +332,7 @@ export const actions: ActionTree<SurveyAndProgramState, RootState> & SurveyAndPr
     async setSurveyResponse(context: ActionContext<SurveyAndProgramState, RootState>, response: SurveyResponse) {
         const {commit, rootState} = context;
         const shapeData = rootState.baseline.shape;
-        if (shapeData) {
+        if (shapeData?.data) {
```

Comment:
I was also occasionally getting errors from shapeData.data.features being null in the call to getDataWithAreaLevel. So making this check cover that case too.